src/kudu/consensus/consensus_queue.cc

```diff
@@ -883,6 +887,97 @@ void PeerMessageQueue::AdvanceQueueWatermark(const char* type,
   }
 }
 
+void PeerMessageQueue::AdvanceMajorityReplicatedWatermark(
```

Comment:
can we put  a comment on what locks are being held while this is being called.
Same comment goes for other places, where such a comment would help.